engine/engine.go

```diff
@@ -89,6 +90,12 @@ type Engine interface {
 	// GetRegistry returns registry with the supported plugins. It should be stored by Engine instance.
 	GetRegistry() *plugin.Registry
 
+	// The Engine can provide a CacheProvider, where it can offer to
+	// cache values in whatever storge the engine itself is backed by.
+	// This would allow especially etcd, but any future backend stores
+	// such as Consul, to cache values.
+	GetCacheProvider() cacheprovider.CacheProvider
```

Comment:
Okay! That works for me actually. Since in our implementation, I'd almost always only use an explicitly provided one. I'll do this too. Due to original discussion, I wanted to reuse engine, but that breaks several principles of proper decoupling. Thanks for the feedback!